src/library/scala/collection/mutable/ArrayBuffer.scala

```diff
@@ -57,7 +57,12 @@ class ArrayBuffer[A] private (initialElements: Array[AnyRef], initialSize: Int)
 
   /** Reduce length to `n`, nulling out all dropped elements */
   private def reduceToSize(n: Int): Unit = {
-    RefArrayUtils.nullElems(array, n, size0)
+    val reducedSize = Math.min(n, size0)
+    val reduceLength = Math.max(initialElements.length, reducedSize)
+    if (array.length > reducedSize * 2)
+      RefArrayUtils.ensureSize(array, reducedSize, reduceLength)
```

Comment:
I'm guessing this should be `array = RefArrayUtils.ensureSize(array, reducedSize, reduceLength)`

Reply:
Good catch!